src/views/studio/MyVideosView/MyVideosView.tsx

```diff
@@ -0,0 +1,180 @@
+const INITIAL_VIDEOS_PER_ROW = 4
+const ROWS_AMOUNT = 4
+// not yet doable
+// TODO: on edit video callbacks
+// TODO: on delete video callbacks
+// TODO: dynamic channels (not hardcoded)
+export const MyVideosView = () => {
+  const [videosPerRow, setVideosPerRow] = useState(INITIAL_VIDEOS_PER_ROW)
+  const [hasAnyVideos, setHasAnyVideos] = useState<boolean>()
+  // Drafts calls can run into race conditions
+  const { drafts, removeDraft, removeAllDrafts, addDraft } = useDrafts('video', testChannelId)
+  const [currentTab, setCurrentTab] = useState(0)
+  const { currentPage, setCurrentPage } = usePagination(currentTab)
+  const videosPerPage = ROWS_AMOUNT * videosPerRow
+  const currentTabName = TABS[currentTab]
+  const isPublic_eq = getPublicness(currentTabName)
+  const { loading, videos, error, totalCount, fetchMore } = useVideosOffsetLimitPagination({
+    limit: videosPerPage,
+    offset: videosPerPage * currentPage,
+    where: {
+      channelId_eq: testChannelId,
+      isPublic_eq,
+    },
+  })
+  useEffect(() => {
+    if ((typeof totalCount === 'number' && totalCount > 0) || drafts.length > 0) {
+      setHasAnyVideos(true)
+    } else if (typeof totalCount === 'number' && hasAnyVideos === undefined) {
+      setHasAnyVideos(false)
+    }
+  }, [drafts.length, hasAnyVideos, totalCount])
+
+  // hook to tests draft should be deleted before final
+  useEffect(() => {
+    const createDrafts = async () => {
+      await removeAllDrafts()
+      for (let i = 0; i < 32; i++) {
+        await addDraft({
+          channelId: testChannelId,
+          title: i.toString(),
+          description: 'string',
+          isPublic: true,
+          hasMarketing: false,
+          isExplicit: false,
+        })
+      }
+    }
+    createDrafts()
+    return () => {
+      removeAllDrafts()
+    }
+  }, [])
+
+  const isDraftTab = currentTabName === 'Drafts'
+  const isLoading = loading || (videos?.length === 0 && (totalCount ?? 0) > 0)
+  const placeholderItems = Array.from({ length: isLoading ? videosPerPage : 0 }, () => ({
+    id: undefined,
+    progress: undefined,
+  }))
+  const videosWithPlaceholders = [...(videos || []), ...placeholderItems]
+  const handleOnResizeGrid = (sizes: number[]) => setVideosPerRow(sizes.length)
+  const hasAnyVideo = currentTabName === 'All Videos' && totalCount === 0 && drafts.length === 0
+
+  const handleChangePage = (page: number) => {
+    setCurrentPage(page)
+    currentTabName !== 'Drafts' &&
+      fetchMore({
+        variables: {
+          offset: videosPerPage * page,
+          limit: videosPerPage,
+        },
+      })
+  }
+
+  const gridContent = (
+    <>
+      {isDraftTab
+        ? drafts
+            // pagination slice
+            .slice(videosPerPage * currentPage, currentPage * videosPerPage + videosPerPage)
+            .map((draft, idx) => (
+              <VideoPreviewPublisher
+                key={idx + '-' + currentTabName + '-' + currentPage}
```

Comment:
Same as the other comment about the `currentTabName` (although maybe it's different because those are drafts).

Also, should we include the `currentPage` in the key? If we do then the components won't get reused right?

Reply:
same as above, I remove these